### PR TITLE
PP-1079 store transactionIds in charge_events table

### PR DESCRIPTION
**Pull request by kakumara**
## WHAT
Smartpay issue a new `pspReference` (maps transactionId in GOV.UK Pay) for every request. Given we have only a single transactionId column in charges, we loose the `pspReference` issued for AUTHORISED when a CAPTURE is completed as we are overriding the value.This PR introduces a new column in charge_events to store transaction ID for every state change so there will be separate rows for each state change.

### CODE REVIEW

**Comment by PauloPortugal** [comment]
@kakumara this is more a general comment, not specific to this PR.
In our team we tend to do a lot of cosmetic refactoring, which should be captured in its own commit to avoid "polluting" the PR/story requirements.
This makes the life of the code reviewer a lot easier, history is cleaner and possibly roll backs (God forbid we have one!) are more straightforward.
What is your opinion on the matter?

File: src/main/java/uk/gov/pay/connector/model/domain/ChargeEventEntity.java [hunk 1]
```diff
@@ -18,6 +18,9 @@
     @Convert(converter = ChargeStatusConverter.class)
     private ChargeStatus status;
 
+    @Column(name = "gateway_transaction_id")
+    private String gatewayTransctionId;
```
Comment (PauloPortugal) [inline src/main/java/uk/gov/pay/connector/model/domain/ChargeEventEntity.java]: Could you please fix the typo **"gatewayTransctionId"** to **"gatewayTransactionId"**?

**Comment by kakumara** [comment]
I completely agree 👍 .  I do find it easier as well. 
I have this habit of spot refactoring when I touch a bit of code and forget to mark them separately. - 

- Can I make few changes more prominent? (One i can see is the renaming of EvenDao to ChargeEventDao, which actually was a feedback from @heathd) 
- anything else you see significant?

File: src/test/resources/templates/smartpay/notification-refund.json [hunk 2]
```diff
@@ -9,11 +9,11 @@
         },
         "eventCode": "REFUND",
         "eventDate": "2015-10-08T13:48:30+02:00",
-        "merchantAccountCode": "DCOEcomm",
-        "merchantReference": "{{transactionId}}-merchant",
-        "originalReference": "{{transactionId}}",
+        "merchantAccountCode": "DCOTest",
```
Comment (PauloPortugal) [inline src/test/resources/templates/smartpay/notification-refund.json]: :+1:

**Review by PauloPortugal — COMMENTED** [review]
Looks OK

**Comment by PauloPortugal** [comment]
@kakumara you do not need to put the changes in another commit for this story, unless you really want. This is more a future action that we should keep in the back of our minds, possibly we should discuss it in a TechRospective (we used to have those) or in a Retrospective. :+1: but apart from the typo, I am happy to approve the PR

File: src/main/java/uk/gov/pay/connector/model/domain/ChargeEventEntity.java (hunk 1 above)
Comment (kakumara) [reply src/main/java/uk/gov/pay/connector/model/domain/ChargeEventEntity.java]: 👍  addressed

**Pull request merged**